### ISSUE TITLE
How should we specify what labels can be provided for training?

### ISSUE THREAD

**Issue by FynnBe** [bioimage-io/python-bioimage-io#19]
Not all tensors/channels that are returned by a (neural network) model require a target to compute a loss. 
Concretely, how does a consumer know what labels to inquire from a user and how to send them?

Three ideas to hopefully kick off further discussion:
As this is clearly linked to model output, a with-`outputs`-integrated suggestion:
```yaml
outputs:
  - name: logits # (2/3 channels should receive labels in this example)
    axes: bcyx
    ...
    shape: [1, 3, 128, 128]
    channels_to_annotate: [membrane, null, nuclei]  # <<< list option
    channels_to_annotate: {0: membrane, 2: nuclei} # <<< dict option
  - name: auxillary or logging output  # (should not receive any labels in this example)
    axes: bc
    ...
    shape: [1, 3]
    channels_to_annotate: null # <<< or [null, null, null] 
```
Problems:
 - How would different output tensors 'share' annotations? (Via identical label names?)
 - What if there are target labels that don't have directly corresponding inputs?

a draft within `training`:
```yaml
training:
    ...
    annotations:
      - name: membrane|nuclei
         axes: bcyx
         shape: [1, 2, 128, 128]
         ...
```

or could we just indicate raw data vs target in the reader output?
in some.reader.yaml:
```yaml
outputs:
  - name: raw
    ...
    is_target: false  # <<< default: false ?
  - name: membrane|nuclei  # <<< '|' could denote channel semantic
     ...
     is_target: true
  - name[s]: [membrane, nuclei] # <<< 1 name per channel
     ...
     is_target: true

```
only (reasonable) limitation here: Readers cannot mix raw and target data within one output array.

**Comment by constantinpape** [comment]
I think this is also outdated.
